reV/config/cli_project_points.py

```diff
@@ -0,0 +1,190 @@
+        of single set of coordinates to convert to ProjectPoints
+    """
+    lat_lons = None
+    msg = None
+    if lat_lon_fpath is not None:
+        lat_lons = lat_lon_fpath
+
+    if lat_lon_coords is not None:
+        if lat_lons is None:
+            lat_lons = lat_lon_coords
+        else:
+            msg = ("Both a 'lat-lon-fpath' and a pair of 'lat-lon-coords' "
+                   "were supplied, but only one can be used to create "
+                   "ProjectPoints!")
+
+    if lat_lons is None:
+        msg = ("A 'lat-lon-fpath' or a pair of 'lat-lon-coords' must be "
+               "supplied in order to create ProjectPoints!")
+
+    if msg is not None:
+        logger.error(msg)
+        raise ProjectPointsValueError(msg)
+
+    return lat_lons
+
+
+def _parse_regions(regions, region, region_col):
+    """
+    Parse CLI regions inputs for ProjectPoints
+
+    Parameters
+    ----------
+    regions : str
+        json string or file path to .json containing regions of
+        interest in the form {'region': 'region_column'}
+    region : str
+        Region to extract
+    region_col : str
+        Meta column to search for region
+
+    Returns
+    -------
+    regions : dict
+        Dictionary of region_col: regions to generate project points
+    """
+    msg = None
+    if regions is not None:
+        if regions.endwtih('.json'):
+            regions = safe_json_load(regions)
+        else:
+            regions = dict_str_load(regions)
+
+    if region is not None:
+        if regions is None:
+            regions = {region: region_col}
+        else:
+            if region in regions:
+                msg = ("Multiple values for {}: {} were provided!"
+                       .format(region, region_col))
+            else:
+                regions.update({region: region_col})
+
+    if regions is None:
+        msg = ("At least a single 'region' and 'region-col' must be "
+               "supplied in order to create ProjectPoints!")
+
+    if msg is not None:
+        logger.error(msg)
+        raise ProjectPointsValueError(msg)
+
+    return regions
+
+
+@click.group()
+@click.option('--fpath', '-f', type=click.Path(), required=True,
+              help='.csv file path to save project points to')
+@click.option('--res_file', '-rf', required=True,
+              help=('Filepath to single resource file, multi-h5 directory, '
+                    'or /h5_dir/prefix*suffix.'))
+@click.option('--tech', '-t', type=STR, default=None,
+              help='reV tech to analyze (required).')
+@click.option('--sam_files', '-sf', required=True, type=SAMFILES,
+              help='SAM config files (required) (str, dict, or list).')
+@click.option('--curtailment', '-curt', type=click.Path(exists=True),
+              default=None,
+              help=('JSON file with curtailment inputs parameters. '
+                    'Default is None (no curtailment).'))
```

Comment:
Why is this needed to make a project points csv file? Maybe ill found out later...